lib/marc_media_type_processor.rb

```diff
@@ -1,114 +1,64 @@
 # frozen_string_literal: true
 
-# A tool for classifying MARC records using a combination of data from the
-# record's leader and some 949ts to assign media types to records
+# A tool for classifying MARC records using a combination of data from the record's leader and some 949ts to assign
+# media types to records
 class MarcMediaTypeProcessor
-  def initialize
-    freeze
```

Comment:
@awead fwiw we had seen concurrency issues so ended up addind freeze to make it stateless and thread-safe, related PR: https://github.com/psu-libraries/psulib_traject/pull/122/files. So, I just wondered about that you haven't seen such behavior?

Reply:
Interesting! Yes, you would run into thread issues with JRuby. I think we can make it thread-safe and refactor at the same time. Let me update it.